src/Bridge/Symfony/Routing/ApiLoader.php

```diff
@@ -111,6 +111,8 @@ public function load($data, $type = null): RouteCollection
                             'property' => $operation['property'],
                             'identifiers' => $operation['identifiers'],
                             'collection' => $operation['collection'],
+                            'parent_resource_class' => $resourceClass,
```

Comment:
We can't say that it's the "parent" because there can be an "infinity" IMO.
Anyway, we need to discuss here is whether we take the "Root" as the metadata entrypoint or the "Queue" (last).
Explanation:

```
/foo/1/bars/2
```

This path will get you an entity of type `Bar`. When this is serialized, where should we look to get the `normalization_context`?
- `Bar` metadata, after all this is a `Bar` response
- `Foo` metadata, because it's a subresource

We need to choose one. At the moment, IIRC the code looks on the `Bar` metadata to find the `normalization_context` for the given operation. 

If we want to take the "Root" (because this is where we declared the subresource), I don't think that you need the "parent_resource_class".
Take a look at the SubresourceOperationFactory: 

https://github.com/api-platform/core/blob/a5dc79534f19db6359d6fa9b14c1e9d07434392c/tests/Operation/Factory/SubresourceOperationFactoryTest.php#L84

`$rootClass = $identifiers[0][1]` is where you want to look at.
the `operation_name` should be the correct one IMO

Reply:
I vote for:
> Foo metadata, because it's a subresource

---

> $rootClass = $identifiers[0][1] is where you want to look at.

yes I noticed that. But I thought it was risky to use that ;)

Reply:
> yes I noticed that. But I thought it was risky to use that ;)

It is not and it's meant for that ;)